dhctl/pkg/kubernetes/actions/converge/converge.go

```diff
@@ -152,13 +154,114 @@ func (r *Runner) isSkip(phase Phase) bool {
 
 func (r *Runner) RunConverge() error {
 	if r.lockRunner != nil {
-		return r.lockRunner.Run(r.converge)
-	} else {
-		return r.converge()
+		err := r.lockRunner.Run()
+		if err != nil {
+			return fmt.Errorf("failed to start lock runner: %w", err)
+		}
+	}
+
+	return r.converge()
+}
+
+func (r *Runner) converge() (err error) {
+	state, err := r.stateStore.GetState()
+	if err != nil {
+		return fmt.Errorf("failed to get converge state: %w", err)
 	}
+
+	r.state = state
+
+	if r.state.NodeUserCredentials == nil {
```

Comment:
Unfortunately, we cannot use this code here because this method also using in autoconverger.

For autoconverger we need converge base infra only. we do not need creating node user and other routines

I think, we need keep old logic of `RunConverge` method, move logic in controller and/or maybe create separate controller for master nodegroup